core/arch/arm/sm/sm.c

```diff
@@ -26,13 +27,17 @@ bool sm_from_nsec(struct sm_ctx *ctx)
 	COMPILE_TIME_ASSERT(!(offsetof(struct sm_ctx, nsec.r0) % 8));
 	COMPILE_TIME_ASSERT(!(sizeof(struct sm_ctx) % 8));
 
-	if (!sm_platform_handler(ctx))
-		return false;
+#ifdef CFG_SM_PLATFORM_HANDLER
+	if (sm_platform_handler(ctx) == SM_HANDLER_SMC_HANDLED)
+		return SM_EXIT_TO_NON_SECURE;
+#endif
 
 #ifdef CFG_PSCI_ARM32
```

Comment:
Stylish question: prefer these `#ifdef` here or in header file with `static inline` definition of stubbed functions?
(question applies to `sm_platform_handler()` and `smc_std_handler()`)

Reply:
In this case I don't think it matters much. I don't mind the `#ifdef`, in this case I think it's easier to see what's going on.

Reply:
ok, thanks for feedback.